employees/views.py

```diff
@@ -156,7 +171,18 @@ def post(self, request: HttpRequest) -> Response:
         )
 
 
-class ReportDetail(APIView):
+@method_decorator(login_required, name="dispatch")
+@method_decorator(
+    check_permissions(
+        allowed_user_types=[
+            CustomUser.UserType.EMPLOYEE.name,
+            CustomUser.UserType.MANAGER.name,
+            CustomUser.UserType.ADMIN.name,
+        ]
+    ),
+    name="dispatch",
+)
+class ReportDetail(UserIsManagerOfCurrentReportProjectMixin, UserIsAuthorOfCurrentReportMixin, APIView):
```

Comment:
@rwrzesien I've checked that this isn't working as we assupted. Can this work if this is still rest framework view? It even does not use `get_queryset` method.

Reply:
That's right, but this view is now refactored so it should work just fine.